horovod/ray/runner.py

```diff
@@ -219,96 +171,59 @@ def create_settings(cls,
             timeout_s=timeout_s,
             placement_group_timeout_s=placement_group_timeout_s)
 
-    def __init__(self,
-                 settings,
-                 num_hosts: int = 1,
-                 num_slots: int = 1,
-                 cpus_per_slot: int = 1,
-                 use_gpu: bool = False,
-                 gpus_per_slot: Optional[int] = None):
-
-        if gpus_per_slot and not use_gpu:
-            raise ValueError("gpus_per_slot is set, but use_gpu is False. "
-                             "use_gpu must be True if gpus_per_slot is set. ")
-        if use_gpu and isinstance(gpus_per_slot, int) and gpus_per_slot < 1:
+    def __init__(
+            self,
+            settings,
+            num_workers: Optional[int] = None,
+            num_hosts: Optional[int] = None,
+            num_workers_per_host: int = 1,
+            cpus_per_worker: int = 1,
+            use_gpu: bool = False,
+            gpus_per_worker: Optional[int] = None,
+            # Deprecated Args.
+            num_slots: Optional[int] = None,
+            cpus_per_slot: Optional[int] = None,
+            gpus_per_slot: Optional[int] = None):
+
+        if num_slots:
+            raise DeprecationWarning("`num_slots` is now deprecated. Please "
+                                     "use the `num_workers` API, "
+                                     "or to enforce an equal number of "
+                                     "workers on each node, set "
+                                     "`num_hosts` and `num_workers_per_host`")
+        if cpus_per_slot or gpus_per_slot:
+            raise DeprecationWarning("`cpus_per_slot` and `gpus_per_slot` "
+                                     "have been deprecated. Use "
+                                     "`cpus_per_worker` and "
+                                     "`gpus_per_worker` instead.")
```

Comment:
for 1 release, can we make this a soft warning (warnings.warn) and then in the later release make this a hard warning?

This will make it much easier to enable migration,